### PR TITLE
fix: Serializers for SecureStorage and Language

### PR DESCRIPTION
**Pull request by satlead**
## What
1. Need validations for SecureStorage api(s) w.r.t Key
2. Need validations for Localization.Language api set properties
3. Need validations for Discovery watchedInfo and EntitlementsData

## Why

To enforce the firebolt schemas specified in the OpenRPC.

## How

By creating custom serializers which make sure that the incoming data is fully compliant to the RegEx specification.
RegEx used in the code matches the definition in the Open RPC.

## Test
Trigger calls for SecureStorage like Get,Set and Remove to validate the key specification.
Trigger Firebolt set call for Localization.language
Trigger Firebolt request for Discovery.contentAccess

## Checklist

- [x] I have self-reviewed this PR
- [x] I have added tests that prove the feature works or the fix is effective

### CODE REVIEW

File: core/sdk/src/api/device/device_request.rs [hunk 1]
```diff
@@ -197,12 +198,15 @@ pub struct OnInternetConnectedRequest {
...
+    #[serde(with = "language_code_serde")]
     pub value: String,
 }
 
 #[derive(Debug, Serialize, Deserialize, Default, Clone)]
 pub struct TimezoneProperty {
+    // Original Regex in the Firebolt Timezone openrpc spec seems to be allowing
+    // even blank strings so the below test case is failing leaving it here
+    // so we can get a future resolution
     //#[serde(with = "timezone_serde")]
```
Comment (Vinodsathyaseelan) [inline core/sdk/src/api/device/device_request.rs:210]: So we are not doing strict timezone format validation here. Is the test cases still failing?

File: core/sdk/src/api/firebolt/fb_discovery.rs [hunk 2]
```diff
@@ -61,10 +63,19 @@ impl LaunchRequest {
 #[derive(Debug, Serialize, Deserialize, Clone, Default)]
 #[serde(rename_all = "camelCase")]
 pub struct EntitlementData {
+    #[serde(deserialize_with = "valid_string_deserializer")]
     pub entitlement_id: String,
```
Comment (Vinodsathyaseelan) [inline core/sdk/src/api/firebolt/fb_discovery.rs:67]: Is empty string not a valid input value for `entitlement_id` ?  Content Access is mainly a pass through feature. 
What if one of the distributors would like  to collect empty string as `entitlement_id` for some specific reason.

File: core/sdk/src/api/device/device_request.rs (hunk 1 above)
Comment (satlead) [reply core/sdk/src/api/device/device_request.rs:210]: Have used the Regex in the openrpc which is used in `serde_utils` it allows a blank timezone to be passed incidentally through the unit test. For now, let this code sit here while we fix the spec eventually we will uncomment it and have it working

File: core/sdk/src/api/firebolt/fb_discovery.rs (hunk 2 above)
Comment (satlead) [reply core/sdk/src/api/firebolt/fb_discovery.rs:67]: Am not sure too feels odd to not have an id given that start_time and end_time are optional too.

Comment (satlead) [reply core/sdk/src/api/firebolt/fb_discovery.rs:67]: will revert this change

**Review by Vinodsathyaseelan — APPROVED** [review]
LGTM

**Review by Vinodsathyaseelan — APPROVED** [review]

**Review by Vinodsathyaseelan — APPROVED** [review]

**Review by Nv-Delle1416 — APPROVED** [review]

**Pull request merged**